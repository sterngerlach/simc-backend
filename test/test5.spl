
{
    int[11] resultTable;
    int i = 0;

    int powerOfTwo(int exponent) 
    {
        int result = 1;

        while ((exponent -= 1) >= 0)
            result *= 2;

        return result;
    }

    int log2(int x)
    {
        int result = 0;

        while ((x /= 2) > 0)
            result += 1;

        return result;
    }

    new(resultTable);

    while (i < 11) {
        resultTable[i] = powerOfTwo(i);
        i += 1;
    }

    for (i = 0 .. 11) {
        sprint("2 ^ ");
        iprint(i);
        sprint(" = ");
        iprint(resultTable[i]);
        sprint(", log2(");
        iprint(resultTable[i]);
        sprint(") = ");
        iprint(log2(resultTable[i]));
        sprint("\n");
    }
}

