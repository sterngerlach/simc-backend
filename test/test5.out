2 ^ 0 = 1, log2(1) = 0
2 ^ 1 = 2, log2(2) = 1
2 ^ 2 = 4, log2(4) = 2
2 ^ 3 = 8, log2(8) = 3
2 ^ 4 = 16, log2(16) = 4
2 ^ 5 = 32, log2(32) = 5
2 ^ 6 = 64, log2(64) = 6
2 ^ 7 = 128, log2(128) = 7
2 ^ 8 = 256, log2(256) = 8
2 ^ 9 = 512, log2(512) = 9
2 ^ 10 = 1024, log2(1024) = 10
